llpc/lower/llpcSpirvLowerRayQuery.cpp

```diff
@@ -594,6 +594,55 @@ Value *SpirvLowerRayQuery::getDispatchId() {
   return dispatchId;
 }
 
+// =====================================================================================================================
+// Create instructions to get BVH SRD given the expansion and box sort mode at the current insert point
+//
+// @param expansion : Box expansion
+// @param boxSortMode : Box sort mode
+Value *SpirvLowerRayQuery::createGetBvhSrd(llvm::Value *expansion, llvm::Value *boxSortMode) {
+  const auto *rtState = m_context->getPipelineContext()->getRayTracingState();
+  assert(rtState->bvhResDesc.dataSizeInDwords == 4);
+
+  // Construct image descriptor from rtstate.
+  Value *bvhSrd = UndefValue::get(FixedVectorType::get(m_builder->getInt32Ty(), 4));
+  bvhSrd =
+      m_builder->CreateInsertElement(bvhSrd, m_builder->getInt32(rtState->bvhResDesc.descriptorData[0]), uint64_t(0));
+  bvhSrd = m_builder->CreateInsertElement(bvhSrd, m_builder->getInt32(rtState->bvhResDesc.descriptorData[1]), 1u);
+  bvhSrd = m_builder->CreateInsertElement(bvhSrd, m_builder->getInt32(rtState->bvhResDesc.descriptorData[2]), 2u);
+  bvhSrd = m_builder->CreateInsertElement(bvhSrd, m_builder->getInt32(rtState->bvhResDesc.descriptorData[3]), 3u);
+
+  Value *bvhSrdDw1 = m_builder->CreateExtractElement(bvhSrd, 1);
```

Comment:
Just use `rtState->bvhResDesc.descriptorData[1]`, saving one needless pair of `insertelement` + `extractelement`.